core/src/test/java/org/elasticsearch/action/DocWriteResponseTests.java

```diff
@@ -51,6 +52,38 @@ public void testGetLocation() {
         assertEquals("/index/type/id?routing=test_routing", response.getLocation("test_routing"));
     }
 
+    public void testGetLocationNonAscii() throws URISyntaxException {
+        DocWriteResponse response =
+            new DocWriteResponse(
+                new ShardId("index", "uuid", 0),
+                "type",
+                "❤",
+                SequenceNumbersService.UNASSIGNED_SEQ_NO,
+                0,
+                Result.CREATED) {
+            };
+        assertEquals("/index/type/%E2%9D%A4", response.getLocation(null));
+        assertEquals("/index/type/%E2%9D%A4?routing=%C3%A4", response.getLocation("%C3%A4"));
+    }
+
+    public void testInvalidGetLocation() {
+        DocWriteResponse invalid =
+            new DocWriteResponse(
+                new ShardId("index", "uuid", 0),
+                "type",
+                "!^*$(@!^!#@",
+                SequenceNumbersService.UNASSIGNED_SEQ_NO,
+                0,
+                Result.CREATED) {
+            };
+        try {
+            invalid.getLocation(null);
+            fail("Did not correctly throw exception on invalid Location parameters");
```

Comment:
I'd use `expectThrows` here. I also like assertion *something* about the exception's massage just to make sure I caught the one I expected.